immuneML/reports/train_ml_model_reports/ReferenceSequenceOverlap.py

```diff
@@ -71,37 +72,36 @@ def build_object(cls, **kwargs):
         return ReferenceSequenceOverlap(**kwargs)
 
     def __init__(self, reference_path: Path = None, comparison_attributes: list = None, name: str = None, state: TrainMLModelState = None,
-                 result_path: Path = None, label: str = None):
-        super().__init__(name)
+                 result_path: Path = None, label: Label = None, number_of_processes: int = 1):
+        super().__init__(name=name, state=state, label=label, result_path=result_path, number_of_processes=number_of_processes)
         self.reference_path = reference_path
         self.comparison_attributes = comparison_attributes
-        self.state = state
-        self.result_path = result_path
-        self.label = label
 
     def _generate(self) -> ReportResult:
 
         figures, tables = [], []
 
         PathBuilder.build(self.result_path)
 
-        if ReferenceSequenceOverlap._check_encoder_class(self.state.optimal_hp_items[self.label].encoder):
+        if ReferenceSequenceOverlap._check_encoder_class(self.state.optimal_hp_items[self.label.name].encoder):
             figure, data = self._compute_optimal_model_overlap()
             figures.append(figure)
             tables.append(data)
 
         for assessment_state in self.state.assessment_states:
-            encoder = assessment_state.label_states[self.label].optimal_assessment_item.encoder
+            encoder = assessment_state.label_states[self.label.name].optimal_assessment_item.encoder
             if ReferenceSequenceOverlap._check_encoder_class(encoder):
-                figure_filename = self.result_path / f"assessment_split_{assessment_state.split_index + 1}_model_vs_reference_overlap_{self.label}.pdf"
-                df_filename = self.result_path / f"assessment_split_{assessment_state.split_index + 1}_overlap_sequences_{self.label}"
+                figure_filename = self.result_path / f"assessment_split_{assessment_state.split_index + 1}_model_vs_reference_overlap_{self.label.name}.pdf"
+                df_filename = self.result_path / f"assessment_split_{assessment_state.split_index + 1}_overlap_sequences_{self.label.name}"
                 figure, data = self._compute_model_overlap(figure_filename, df_filename, encoder,
                                                            f"overlap sequences between the model for assessment split "
                                                            f"{assessment_state.split_index + 1} and reference list")
                 figures.append(figure)
                 tables.append(data)
 
-        return ReportResult(self.name, output_figures=figures, output_tables=tables)
+        return ReportResult(self.name,
+                            info="A Venn diagram between the list of disease-associated sequences produced by the SequenceAbundance encoder and a list of reference receptor sequences, and a file containing the overlapping sequences.",
```

Comment:
could you make sure that this info is in the same place for all reports? now it is sometimes an object variable, sometimes it's a constant hard-coded when constructing ReportResult object :)